psm_utils/io/idxml.py

```diff
@@ -179,17 +187,11 @@ def _parse_psm(
             # Storing proforma notation of peptidoform and UNIMOD peptide sequence for mapping back
             # to original sequence in writer
             provenance_data={str(peptidoform): peptide_hit.getSequence().toString()},
-            # TODO: Can we recover any all UserParams and save them as metadata?
-            # This is needed to calculate a qvalue before rescoring the PSMList
-            metadata={
-                "idxml:score_type": str(peptide_id.getScoreType()),
-                "idxml:higher_score_better": str(peptide_id.isHigherScoreBetter()),
-                "idxml:significance_threshold": str(peptide_id.getSignificanceThreshold()),
-            },
+            # Store metadata of PeptideIdentification and PeptideHit objects
+            metadata=peptide_id_metadata | peptide_hit_metadata,
```

Comment:
todo: Nicely done with the union operator! Unfortunately, I believe this is not yet supported in Python 3.8 :( Can you replace it with [{**d1, **d2}](https://peps.python.org/pep-0584/#d1-d2)?

Reply:
Sure!